tests/checkpointing/test_model_checkpoint.py

```diff
@@ -951,30 +955,28 @@ def assert_checkpoint_log_dir(idx):
     model = ExtendedBoringModel()
     trainer.fit(model)
     assert trainer.global_step == epochs * limit_train_batches
-    assert trainer.current_epoch == epochs - 1
+    assert trainer.current_epoch == epochs
     assert_checkpoint_log_dir(0)
     assert_checkpoint_content(ckpt_dir)
 
     trainer.validate(model)
-    assert trainer.current_epoch == epochs - 1
+    assert trainer.current_epoch == epochs
 
     trainer.test(model)
-    assert trainer.current_epoch == epochs - 1
+    assert trainer.current_epoch == epochs
 
     for idx in range(1, 5):
         chk = get_last_checkpoint(ckpt_dir)
         assert_checkpoint_content(ckpt_dir)
 
         # load from checkpoint
-        trainer_config["callbacks"] = [ModelCheckpoint(dirpath=ckpt_dir, save_top_k=-1)]
```

Comment:
Need to remove this because creating a new instance means we lose the existing value for

`ModelCheckpoint._last_global_step_saved`

which avoids saving a duplicate checkpoint on the second fit call. Perhaps this value should be saved and reloaded so that creating a new instance also works.